src/librustc_data_structures/indexed_vec.rs

```diff
@@ -157,7 +157,7 @@ macro_rules! newtype_index {
     (@pub          [$($pub:tt)*]
```

Comment:
As a side note, this could be using `$pub:vis` instead of `$($pub:tt)*`.

Reply:
I could add that as a separate commit.

Reply:
My attempt to make this change https://gist.github.com/spastorino/715079ea4839effe67fd9f0c3bee1c70

And I'm getting a lot of these errors ...
```
error: no rules expected the token `]`
  --> src/librustc/dep_graph/graph.rs:48:1
   |
48 | newtype_index!(DepNodeIndex);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate
```

Ran with trace macros ...

```
note: trace_macro
  --> src/librustc/dep_graph/graph.rs:48:1
   |
48 | newtype_index!(DepNodeIndex);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: expanding `newtype_index! { DepNodeIndex }`
   = note: to `newtype_index ! (
           @ type [ DepNodeIndex ] @ max [ :: std :: u32 :: MAX ] @ debug_format [ "{}" ]
           ) ;`
   = note: expanding `newtype_index! { @ type [ DepNodeIndex ] @ max [ :: std :: u32 :: MAX ] @ debug_format [ "{}" ] }`
   = note: to `newtype_index ! (
           @ vis [  ] @ type [ DepNodeIndex ] @ max [ ::std::u32::MAX ] @ debug_format [
           "{}" ] ) ;`
   = note: expanding `newtype_index! { @ vis [  ] @ type [ DepNodeIndex ] @ max [ ::std::u32::MAX ] @ debug_format [
           "{}" ] }`

error: no rules expected the token `]`
  --> src/librustc/dep_graph/serialized.rs:17:1
   |
17 | newtype_index!(SerializedDepNodeIndex);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate
```

Reply:
This could be #41949. Probably not worth it to fix in this PR :)